peerdiversity/filter.go

```diff
@@ -220,6 +220,13 @@ func (f *Filter) ipGroupKey(ip net.IP) (PeerIPGroupKey, error) {
 		if err != nil {
 			return "", fmt.Errorf("failed to fetch ASN for IPv6 addr %s: %w", ip.String(), err)
 		}
+
+		// if no ASN found then fallback on using the /32 prefix
+		if len(s) == 0 {
+			dfLog.Debugw("group key is empty", "appKey", f.logKey, "ip", ip.String())
```

Comment:
```suggestion
			dfLog.Debugw("ASN not known", "ip", ip)
```

?

Reply:
why take out the appKey? It's not particularly relevant to the error, however, I suspect people may just grep for the appKey and then will miss these.

Reply:
Can take out the `.String()`. I've been a bit scarred by dealing with PeerIDs and the constant casting between []byte, string, and PeerID, so I'm always throwing in `.String()` to make sure it's really obvious that we're getting out a pretty loggable string.

Reply:
> why take out the appKey? It's not particularly relevant to the error, however, I suspect people may just grep for the appKey and then will miss these.

Ah, nevermind. I thought this was somehow related to a query key.

(nit: the constructor should accept a custom logger if we want the user to be able to add fields)

> Can take out the .String(). I've been a bit scarred by dealing with PeerIDs and the constant casting between []byte, string, and PeerID, so I'm always throwing in .String() to make sure it's really obvious that we're getting out a pretty loggable string.

You can take out the `String()` in pretty much all cases. It'll save us some work when debug logging is disabled.